Alerting: Fix display of paused recording rules

tomratcliffe opened the pull request:
**What is this feature?**
Fixes display of paused recording rules on the list view and detail views

**Why do we need this feature?**
(I think, but didn't verify that) A recent refactor in https://github.com/grafana/grafana/pull/101639 caused a small regression. (Tests are to come on this PR as well)

**Who is this feature for?**
Alerting users

Fixes https://github.com/grafana/grafana/issues/102342

tomratcliffe committed:
Fix display of paused recording rules

soniaAguilarPeiron commented on public/app/features/alerting/unified/components/rule-viewer/RuleViewer.tsx:
```diff
@@ -73,7 +73,9 @@ const RuleViewer = () => {
   const isProvisioned = rulerRuleType.grafana.rule(rulerRule) && Boolean(rulerRule.grafana_alert.provenance);
-  const isPaused = rulerRuleType.grafana.alertingRule(rulerRule) && isPausedRule(rulerRule);
+  const isPaused =
+    (rulerRuleType.grafana.alertingRule(rulerRule) || rulerRuleType.grafana.recordingRule(rulerRule)) &&
```
nit: given the logic is used in several places I think we could extract to a separate function so we can re-use it.

gillesdemey replied:
There should be one `rulerRule.grafana.rule()`

gillesdemey commented on public/app/features/alerting/unified/components/rule-viewer/tabs/Details.tsx:
```diff
@@ -74,7 +74,9 @@ export const Details = ({ rule }: DetailsProps) => {
   const updated = rulerRuleType.grafana.rule(rule.rulerRule) ? rule.rulerRule.grafana_alert.updated : undefined;
-  const isPaused = rulerRuleType.grafana.alertingRule(rule.rulerRule) && isPausedRule(rule.rulerRule);
+  const isPaused =
+    (rulerRuleType.grafana.alertingRule(rule.rulerRule) || rulerRuleType.grafana.recordingRule(rule.rulerRule)) &&
```
```suggestion
    (rulerRuleType.grafana.rule(rule.rulerRule) &&
```

gillesdemey commented on public/app/features/alerting/unified/components/rules/RulesTable.tsx:
```diff
@@ -353,7 +353,9 @@ function useRuleStatus(rule: CombinedRule) {
   const isCreating = Boolean(hasRuler && rulerRulesLoaded && rulerRule && !promRule);
-  const isPaused = rulerRuleType.grafana.alertingRule(rulerRule) && isPausedRule(rulerRule);
+  const isPaused =
+    (rulerRuleType.grafana.alertingRule(rulerRule) || rulerRuleType.grafana.recordingRule(rulerRule)) &&
```
dito `rulerRule.grafana.rule()` :)

gillesdemey commented:
> (I think, but didn't verify that) A recent refactor in https://github.com/grafana/grafana/pull/101639 caused a small regression.

I checked and it's a regression :)

gillesdemey committed:
Simplify paused state check for Grafana alert rules

The change eliminates redundant rule type checks by using a more generic
grafana.rule() check that covers both alerting and recording rules.

tomratcliffe committed:
Add test for recording rule paused state

tomratcliffe committed:
Merge branch 'main' into alerting/recording-rule-paused-state

gillesdemey committed:
Add test for paused Grafana alert in Details tab

soniaAguilarPeiron reviewed (approved):
LGTM!
